### PR TITLE
Add a macos static build that (re-)enables securetransport

### PR DESCRIPTION
**Pull request by wolfv**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/install.bat [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+nmake /f Makefile.vc MODE=dll VC=%VS_MAJOR:"=% WITH_DEVEL=%LIBRARY_PREFIX% ^
+         WITH_ZLIB=dll WITH_SSH2=dll DEBUG=no ENABLE_IDN=no ENABLE_SSPI=yes ^
+         MACHINE=%ARCH_STRING% ENABLE_UNICODE=yes
+if errorlevel 1 exit 1
+
+REM This is implicitly using WinSSL.  See Makefile.vc for more info.
+nmake /f Makefile.vc mode=static VC=%VS_MAJOR:"=% WITH_DEVEL=%LIBRARY_PREFIX% ^
+         WITH_ZLIB=dll WITH_SSH2=dll DEBUG=no ENABLE_IDN=no ENABLE_SSPI=yes ^
+         MACHINE=%ARCH_STRING% ENABLE_UNICODE=yes
+if errorlevel 1 exit 1
```
Comment (isuruf) [inline recipe/install.bat:19]: Why do you need to build again?

File: recipe/build.sh [hunk 2]
```diff
@@ -5,22 +5,22 @@ cp $BUILD_PREFIX/share/libtool/build-aux/config.* .
...
+    $USESSL $STATIC_BUILD \
     --with-zlib=${PREFIX} \
     --with-gssapi=${PREFIX} \
     --with-libssh2=${PREFIX} \
     --with-nghttp2=${PREFIX} \
 || cat config.log
 
 make -j${CPU_COUNT} ${VERBOSE_AT}
-# TODO :: test 1119... exit FAILED
-# make test
```
Comment (isuruf) [inline recipe/build.sh:22]: Keep this

Comment (wolfv) [reply recipe/build.sh:22]: this was already commented out but we can try again :)

Comment (wolfv) [reply recipe/build.sh:22]: ah you mean, keep as a comment, sure.

File: recipe/install.bat (hunk 1 above)
Comment (wolfv) [reply recipe/install.bat]: I don't, thanks.

File: recipe/build.sh (hunk 2 above)
Comment (wolfv) [reply recipe/build.sh]: thanks for reviewing!

File: recipe/build.sh [hunk 3]
```diff
@@ -5,22 +5,25 @@ cp $BUILD_PREFIX/share/libtool/build-aux/config.* .
 # need macosx-version-min flags set in cflags and not cppflags
 export CFLAGS="$CFLAGS $CPPFLAGS"
 
+if [[ "${PKG_NAME}" == "libcurl-securetransport-static" ]]; then
+    USESSL="--with-secure-transport"
+    STATIC_BUILD="--enable-static --disable-shared"
+else
+    USESSL="--with-openssl=${PREFIX}"
+fi
```
Comment (isuruf) [inline recipe/build.sh:13]: When this script is called, `PKG_NAME` will always be `curl_split_recipe`

Comment (wolfv) [reply recipe/build.sh:13]: thanks @isuruf, great catch! sorry

**Comment by wolfv** [comment]
@isuruf would you be willing to merge this PR? It would be quite nice to release micromamba 1.0 with updated curl (& libarchive) :)

**Review by isuruf — COMMENTED** [review]
Can't we have a build with both `--with-openssl --with-secure-transport`?

**Comment by wolfv** [comment]
I think that didn't work in the past: 

- you removed `darwin-ssl`: https://github.com/conda-forge/curl-feedstock/pull/91/files 
- because it was crashing git and other tools with older macos versions: https://github.com/conda-forge/curl-feedstock/issues/89

**Comment by isuruf** [comment]
How about the default being `---with-openssl --with-secure-transport -with-default-ssl-backend=openssl` and then `micromamba` can use `secure-transport` by default?

**Comment by wolfv** [comment]
Happy to try that.

**Comment by isuruf** [comment]
`curl_global_sslset` is the function to do that.

**Comment by wolfv** [comment]
yeah, we're using that function already https://github.com/mamba-org/mamba/blob/c5d73a204b19311114885f3081b86e7d881e151e/libmamba/src/core/fetch.cpp#L45-L49

**Comment by wolfv** [comment]
Closing in favor of #112

**Pull request closed without merging**